tests/test_device.py

```diff
@@ -694,3 +709,45 @@ def test_outdated_API(self, monkeypatch):
             m.setattr(qml, "version", lambda: "0.0.1")
             with pytest.raises(DeviceError, match="plugin requires PennyLane versions"):
                 qml.device("default.qubit", wires=0)
+
+
+class TestBatchExecution:
+    """Tests for the batch_execute method."""
+
+    @pytest.mark.parametrize("n_tapes", [1, 2, 3])
+    def test_calls_to_execute(self, n_tapes, mocker, mock_device_with_apply):
+        """Tests that the device's execute method is called the correct number of times."""
+
+        dev = mock_device_with_apply(wires=2)
+        spy = mocker.spy(Device, "execute")
+
+        tapes = [qml.tape.QuantumTape()]*n_tapes
+        dev.batch_execute(tapes)
+
+        assert spy.call_count == n_tapes
```

Comment:
Nice!

Reply:
Although I suppose this is very specific to the `QubitDevice`/`Device` implementation